### PR TITLE
38 BatchController

### LINKED ISSUE #38: BatchController
Autowires a BatchReportService on getBatch and another on getBatchPDFreport

Exposes API endpoint.

### PR DESCRIPTION
**Pull request by Kaszz**
Closes #38 

## Description
Add BatchController

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes, if necessary.
- [x] All tests pass (existing and potential new).
- [x] Sonar check has passed
- [x] No codesmells or bugs

## Other Relevant Information
Also required a new IBatchGetterService and BatchGetterImpl so it was possible to access the database to check if a batch exists.
Tests were also written.

### CODE REVIEW

File: BrewMesModulesParent/api/src/main/java/com/brewmes/api/BatchController.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/batch")
+public class BatchController {
+
+    @Autowired(required = false)
```
Comment (Lawst96) [inline BrewMesModulesParent/api/src/main/java/com/brewmes/api/BatchController.java:18]: Bør vi nok fjerne, mens vi husker det.
```suggestion
    @Autowired
```

File: BrewMesModulesParent/api/src/main/java/com/brewmes/api/BatchController.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    @Qualifier("dashboard")
+    IBatchReportService dashboardService;
+
+    @Autowired(required = false)
```
Comment (Lawst96) [inline BrewMesModulesParent/api/src/main/java/com/brewmes/api/BatchController.java:22]: Også her.
```suggestion
    @Autowired
```

**Review by Wylardt — APPROVED** [review]

**Review by Lawst96 — APPROVED** [review]

File: BrewMesModulesParent/api/src/main/java/com/brewmes/api/BatchController.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    /**
+     * Starts the generation of the PDF report and gets the path for the PDF report
+     * @param id The {@code ID} of the {@code Batch}
+     * @return The {@code ResponseEntity} containing the path for the PDF report download
+     */
+    @GetMapping(value = "/{id}/pdf")
+    public ResponseEntity<String> getBatchPdfReport(@PathVariable("id") String id) {
+        if (getter.containsId(id)) {
+            return new ResponseEntity<>(pdfService.prepareBatchReportService(id), HttpStatus.OK);
```
Comment (dottore1) [inline BrewMesModulesParent/api/src/main/java/com/brewmes/api/BatchController.java:62]: Denne metode bør indeholde mere logik for at vi i frontenden kan downloade den som pdf - Se evt BrewMES controlleren fra sidste semester.

**Review by dottore1 — CHANGES_REQUESTED** [review]

**Comment by Kechr** [comment]
Please fix the code smell. I can see that you have checked the box without checking for code smells beforehand, please refrain from doing this in the future.

File: BrewMesModulesParent/api/src/main/java/com/brewmes/api/BatchController.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+     * Starts the generation of the PDF report and gets the path for the PDF report
+     * @param id The {@code ID} of the {@code Batch}
+     * @return The {@code ResponseEntity} containing the path for the PDF report download
+     */
+    @GetMapping(value = "/{id}/pdf")
+    public ResponseEntity<Object> getBatchPdfReport(@PathVariable("id") String id) {
+        if (getter.containsId(id)) {
+            pdfService.prepareBatchReportService(id);
+
+            String fileName = "batch_report.pdf"; //NOSONAR
```
Comment (dottore1) [inline BrewMesModulesParent/api/src/main/java/com/brewmes/api/BatchController.java:70]:
```suggestion
            String fileName = pdfService.prepareBatchReportService(id); //NOSONAR
```

**Review by dottore1 — CHANGES_REQUESTED** [review]

**Comment by Kechr** [comment]
> > Please fix the code smell. I can see that you have checked the box without checking for code smells beforehand, please refrain from doing this in the future.
> 
> It was from an implemented class that was covered by another issue but will be fixed. :)

Aight, my bad for not catching it the first time. I will up my game 💯

**Review by Kechr — APPROVED** [review]

**Pull request merged**